src/streaming/controllers/ScheduleController.js

```diff
@@ -189,7 +189,7 @@ function ScheduleController(config) {
 
         const isReplacement = replaceRequestArray.length > 0;
         const streamInfo = streamProcessor.getStreamInfo();
-        if (bufferResetInProgress || isNaN(lastInitQuality) || switchTrack || isReplacement ||
+        if (!streamProcessor.getBufferController().getIsPruningInProgress() && bufferResetInProgress || isNaN(lastInitQuality) || switchTrack || isReplacement ||
```

Comment:
Shouldn't we put all the or conditions within parenthesis?

if (!streamProcessor.getBufferController().getIsPruningInProgress() && (bufferResetInProgress || isNaN(lastInitQuality) || switchTrack || isReplacement ||
hasTopQualityChanged(currentRepresentationInfo.mediaInfo.type, streamInfo.id) ||
bufferLevelRule.execute(streamProcessor, streamController.isTrackTypePresent(Constants.VIDEO))))

Reply:
Agreed, that's a good catch 👍